feat/feature.go

```diff
@@ -105,6 +105,24 @@ type Collection interface {
 	Location() Feature
 }
 
+// BasePositionOf returns the position in f converted to coordinates relative
+// to the shallowest nil feature location and a reference feature which will
+// be the deepest non nil feature. The reference feature can be used by
+// different calls to BasePositionOf to verify that returned positions are
+// comparable.
+// BasePositionOf will panic if the feature chain is deeper than 1000 links.
+//
+// or
+//
+// BasePositionOf returns the position in f converted to coordinates relative
+// to the shallowest nil feature location and a reference feature which will
+// be the parent feature of the swallowest nil feature. The reference feature
+// can be used by different calls to BasePositionOf to verify that returned
+// positions are comparable.
+// BasePositionOf will panic if the feature chain is deeper than 1000 links.
+//
+// or
+//
 // BasePositionOf returns the position in f converted to coordinates in the
```

Comment:
"relative to the first nil feature location" is not right intuitively. I'm not sure how to phrase it. Consider the case of a chromosome type that returns nil for Location(). A feature on that chromosome should return coordinates relative to the chromosome, not the nil.